### PR TITLE
Handle torch.seed

### PR DESCRIPTION
**Pull request by anijain2305**
@jansel This is probably the incorrect way, looking for feedback.

The generated Fx graph for 

~~~
        def fn(x):
            attention_seed = int(torch.seed() % sys.maxsize)
            torch.manual_seed(attention_seed)
            return ()
~~~

is 
~~~

def forward(self):
    seed = torch.random.seed()
    mod = seed % 9223372036854775807;  seed = None
    int_1 = int(mod);  mod = None
    manual_seed = torch.random.manual_seed(int_1);  int_1 = None
    return ()

~~~

### CODE REVIEW

File: torchdynamo/variables/builtin.py [hunk 1]
```diff
@@ -94,6 +96,7 @@ def can_constant_fold_through(self):
     @functools.lru_cache(None)
     def _fx_graph_functions():
         fns = {
+            int,
```
Comment (jansel) [inline torchdynamo/variables/builtin.py:99]: Perhaps we should be more selective about admitting `int()` into the FX graph.  If someone calls `int(Tensor)` we should likely break the graph on it.  So maybe allow int on a SeedVariable/DynamicShapeVariable but not a general tensor?

**Review by jansel — CHANGES_REQUESTED** [review]
I think this will cause

File: torchdynamo/variables/builtin.py (hunk 1 above)
Comment (anijain2305) [reply torchdynamo/variables/builtin.py:99]: I agree. I don't like `int()` in the graph. Yes, I think allowing `int` on SeedVariable/DynamicShapeVariable is acceptable.  Will make the changes.

**Comment by anijain2305** [comment]
Ready for review.

**Review by jansel — APPROVED** [review]

**Pull request merged**